Rust/src/common/rangevector.rs

```diff
@@ -43,7 +43,7 @@ impl RangeVector {
     }
 
     pub fn shift(&mut self, i:usize, shift: f32) {
-        for j in 0..self.values.len() {
+        for _ in 0..self.values.len() {
```

Comment:
Yikes, this is a bug! The intention was to change the single coordinate once. This loop should be removed. We can have a separate PR or fix in this one.

Reply:
Great, fixed this and `scale` in a follow-up commit.